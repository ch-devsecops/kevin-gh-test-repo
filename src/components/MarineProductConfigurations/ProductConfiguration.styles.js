import { ThemeStyles } from '@honda-canada/design-system-react';
import css from '@styled-system/css';

const themeStyles = new ThemeStyles();

themeStyles.addBaseStyles('Container')(({ theme }) =>
  css({
    backgroundColor: 'grey.5',
    border: `${theme.borders[1]} ${theme.colors.grey[2]}`,
    display: 'flex',
    alignItems: ['flex-start', 'flex-start', 'center'],
    flexDirection: ['column', 'column', 'row'],
    justifyContent: ['flex-start', 'flex-start', 'space-between'],
    borderRadius: '4px',
    py: 'm',
    px: 'm',
    mt: 0,
    mx: ['m', 'm', 'big'],
    mb: ['xs', 'xs', 'm'],
    ':last-child': {
      mb: ['default', 'default', 'l'],
    },
  }),
);

themeStyles.addBaseStyles('Header')(
  css({
    mb: 'xs',
  }),
);

themeStyles.addBaseStyles('CopyContainer')(
  css({
    width: ['90%', '90%', '75%'],
  }),
);

themeStyles.addBaseStyles('PriceComponentWrapper')(({ hasDiscount }) => {
  const height = hasDiscount ? '92px' : '40px';
  return css({
    width: 'max-content',
    height: '100%',
    display: 'flex',
    alignItems: 'center',
    mb: ['default', 'default', 0],
    mt: ['m', 'm', 0],
    '> div': {
      height: [height, height, undefined],
      flexDirection: 'column',
    },
  });
});

export const priceComponentStyles = {
  pricesContainer: {
    pb: 0,
  },
  priceStyles: {
    copy: {
      styles: {
        fontSize: ['24px', '24px', '24px'],
        lineHeight: ['28px', '28px', '28px'],
      },
    },
    discount: {
      styles: {
        fontSize: ['16px', '16px', '16px'],
      },
    },
  },
  saveStyles: {
    container: {
      mx: 0,
    },
    copy: {
      styles: {
        fontSize: ['14px'],
      },
    },
  },
};

export default themeStyles;
