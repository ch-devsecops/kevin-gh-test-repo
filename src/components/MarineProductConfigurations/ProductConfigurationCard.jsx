import React from 'react';
import { Box, Copy, H5, useMediaQueries } from '@honda-canada/design-system-react';

import PriceComponent from '../PriceComponent';
import { usePricesConfiguration } from '../../utils/financeUtils';
import themeStyles from './ProductConfiguration.styles';

const Container = themeStyles.apply(Box, 'Container');
const CopyContainer = themeStyles.apply(Box, 'CopyContainer');
const Header = themeStyles.apply(H5, 'Header');
const PriceComponentWrapper = themeStyles.apply(Box, 'PriceComponentWrapper');

const priceComponentStyles = {
  saveStyles: {
    container: {
      mx: [0, 0, 0],
    },
  },
};

const ProductConfigurationCard = ({ configuration, productPrice }) => {
  const { isMobile } = useMediaQueries();
  const name = configuration?.configurationName?.value;
  const taglineValue = configuration?.tagline?.value;
  const configurationId = configuration?.detIdentifier?.value;
  const getPrices = usePricesConfiguration();
  const { modelConfiguration, isFetching, hasError } = productPrice;
  const prices = getPrices(modelConfiguration, configurationId);

  return (
    <Container data-testid="configuration-card">
      <CopyContainer isMobile={isMobile}>
        <Header data-testid="configuration-name">{name}</Header>
        <Copy size="regular" data-testid="configuration-tagline">
          {taglineValue}
        </Copy>
      </CopyContainer>
      <PriceComponentWrapper hasDiscount={!!prices?.discount?.priceDiscountAmount}>
        <PriceComponent
          prices={prices}
          horizontalAlignment={['left', 'left', 'right']}
          isFetching={isFetching}
          hasError={hasError}
          priceComponentStyles={priceComponentStyles}
          display="flex"
          saveAboveLabel
          alignItems={['flex-start', 'flex-start', 'flex-end']}
        />
      </PriceComponentWrapper>
    </Container>
  );
};

export default ProductConfigurationCard;
