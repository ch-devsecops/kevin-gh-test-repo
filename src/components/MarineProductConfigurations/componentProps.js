import PropTypes from 'prop-types';

export default {
  fields: PropTypes.shape({
    data: PropTypes.shape({
      value: PropTypes.shape({
        model: PropTypes.shape({
          fields: PropTypes.shape({
            id: PropTypes.string,
            name: PropTypes.string,
            modelName: PropTypes.shape({
              value: PropTypes.string,
            }),
            configurations: PropTypes.arrayOf(
              PropTypes.shape({
                items: PropTypes.arrayOf(
                  PropTypes.shape({
                    configurationName: PropTypes.shape({
                      value: PropTypes.string,
                    }),
                    name: PropTypes.string,
                    tagline: PropTypes.shape({
                      value: PropTypes.string,
                    }),
                    modelCode: PropTypes.shape({
                      value: PropTypes.string,
                    }),
                    detIdentifier: PropTypes.shape({
                      value: PropTypes.string,
                    }),
                  }),
                ),
              }),
            ),
          }),
        }),
      }),
    }),
  }),
  params: PropTypes.shape({
    autoId: PropTypes.string,
    horizontalMargin: PropTypes.string,
    topMargin: PropTypes.string,
    bottomMargin: PropTypes.string,
  }),
  rendering: PropTypes.shape({
    componentName: PropTypes.string,
  }),
};
