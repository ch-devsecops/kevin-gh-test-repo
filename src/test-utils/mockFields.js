/* eslint-disable max-len */
/* eslint-disable max-classes-per-file */

// FIXME: jsdoc popup not working on test files
/** Class representing a mock data. */
class MockData {
  /**
   * Add a field.
   * @param {string} key - key of the field obj: i.e. 'title'
   * @param {Object} value - value of the field obj: i.e. { value: 'some text' }.
   */
  add(key, value) {
    this[key] = value;
  }
}

/**
 * Class representing mock fields object.
 * @extends MockData
 *
 * available method: add()
 *
 * default fields: [anchorId, title, bodyText, image, ctaLink, ctaType, ctaIcon, gtmCategory, gtmTitle, contentTitle, subtitle]
 *
 * note: we can add more commonly used fields in the class declaration if needed
 */
class MockFields extends MockData {
  constructor() {
    super();

    this.title = { value: 'Title' };
    this.bodyText = { value: 'BodyText' };
    this.image = {
      value: { src: 'mock.png', alt: 'mock alt' },
    };
    this.ctaLink = {
      value: {
        href: 'www.source.ca',
        title: 'cta aria label',
        text: 'CTA Text',
      },
    };
    this.ctaType = {
      value: 'Tertiary',
    };
    this.ctaIcon = {
      fields: {
        value: {
          value: 'arrowRight',
        },
      },
    };
    this.gtmCategory = {
      fields: {
        value: {
          value: 'Others',
        },
      },
    };
    this.gtmTitle = {
      value: 'gtm title',
    };
    this.contentTitle = {
      value: 'Content Title',
    };
    this.subtitle = {
      value: 'Content Subtitle',
    };
    this.seoH1 = {
      value: '',
    };
    this.anchorId = {
      value: 'anchor-id',
    };
    this.ctaLabel = {
      value: 'CTA Label',
    };
  }
}

/**
 * Class representing a mock rendering object
 * @extends MockData
 *
 * available method: add()
 */
class MockRendering extends MockData {
  /**
   * Create a mock rendering object
   * @param {string} componentName - adds componentName
   */
  constructor(componentName) {
    super();

    this.componentName = componentName;
  }
}

export { MockFields, MockRendering };
