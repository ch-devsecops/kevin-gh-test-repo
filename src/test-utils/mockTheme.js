/**
 * Mock styled-components theme, if you want to test how your component
 * implements theme styles. Import as-is, or spread in any
 * additional values you need.
 *
 * Example usage:

  it('should apply the bgColor param', () => {
    const { colors } = MockTheme;
    const { container } = render(
      <MyComponent
        fields={mockFields}
        params={{ bgColor: 'Gray' }}
      />,
    );
    const wrapper = container.querySelector('section');

    expect(wrapper).toHaveStyleRule('background-color', colors.grey[5]);
  });
 */

import { HONDA_THEME_NAME } from '../utils/constants';

const mockTheme = {
  letterSpacings: {
    sm: '0.5px',
    md: '1px',
  },
  lineHeights: {
    sm: '22px',
    md: '26px',
  },
  colors: {
    white: '#fff',
    transparent: 'transparent',
    grey: ['#707070', '#C7C7C7', '#D5D5D5', '#EAEAEA', '#F8F8F8', '#F5F5F5'],
    black: '#2C2C2C',
    red: '#DA2C2E',
    redShades: ['rgba(200, 16, 46, 0.1)'],
    aodaRed: '#D92629',
    lightRed: '#EC9597',
    darkRed: '#A21C1D',
    blue: '#334C96',
    darkBlue: '#002D62',
    lightBlue: '#00447826',
    yellow: ['#F9CA52', '#EDAB00', '#8D763F'],
    fadedBlue: '#ADB7D5',
    primary: '#DA2C2E',
    primaryAoda: '#D92629',
    primaryDark: '#A21C1D',
    primaryLight: '#EC9597',
    secondary: '#2C2C2C',
    secondaryLight: '#C7C7C7',
    tertiary: '#002D62',
    tertiaryLight: '#334C96',
    tertiaryDisabled: '#ADB7D5',
    danger: '#DA2C2E',
    warning: '#2C2C2C',
    warningLight: '#EAEAEA',
    success: '#334C96',
    successLight: '#d9e0e7',
    disabled: '#C7C7C7',
    typographyDefault: '#2C2C2C',
  },
  space: [0, 4, 8, 12, 16, 24, 32, 48, 64, 96, 128, 144, 192, 256],
  borders: [0, '1px solid', '2px solid', '3px solid'],
  breakpoints: ['768px', '992px', '1200px'],
  fontWeights: {
    regular: 'normal',
    bold: 700,
  },
  fontSizes: {
    sm: '12px',
  },
  shadows: {
    smooth: {
      md: '0.1px 0.3px 0.3px rgba(0, 0, 0, 0.02),0.3px 0.7px 0.9px rgba(0, 0, 0, 0.028),0.7px 1.4px 1.8px rgba(0, 0, 0, 0.035),1.5px 2.9px 3.7px rgba(0, 0, 0, 0.047),4px 8px 10px rgba(0, 0, 0, 0.07)',
    },
    box: {
      subNav: '0px 2px 4px rgba(0, 0, 0, 0.1)',
    },
  },
  zIndices: {
    accordionItemTitle: 200,
    modal: 13001,
    sectionNav: 13000,
  },
  transitionDuration: {
    t1: '1s',
  },
  transitionTimingFunction: {
    default: 'ease-in-out',
    out: 'ease-out',
    in: 'ease-in',
  },
  name: HONDA_THEME_NAME,
  fonts: {
    default: 'Helvetica',
    bold: 'Cooper Black',
    heading: 'Roboto',
  },
  header: {
    desktop: {},
    mobile: {},
  },
  icons: {},
  logIconWarnings: false,
};

export default mockTheme;
