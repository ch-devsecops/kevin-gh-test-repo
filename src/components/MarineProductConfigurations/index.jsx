import React, { useEffect } from 'react';
import ProductConfigurationCard from './ProductConfigurationCard';
import componentProps from './componentProps';
import { MODEL_YEAR_API_DEFAULT_VALUE } from '../../utils/constants';
import useProductFinancial from '../../utils/hooks/useProductFinancial';
import { makeModelConfiguration } from '../../utils/financeUtils';

const MarineProductConfigurations = ({ fields }) => {
  const data = fields?.data?.value?.model?.fields;
  const configurations = data?.configurations?.[0]?.items;
  const detModelKey = data?.detModelKey?.value;
  const {
    financial: { modelConfiguration },
    isFetching,
    hasError,
    setModels,
  } = useProductFinancial({ formatter: makeModelConfiguration });

  useEffect(() => {
    if (detModelKey) {
      setModels([
        {
          modelKey: detModelKey,
          modelYear: MODEL_YEAR_API_DEFAULT_VALUE,
        },
      ]);
    }
  }, [detModelKey]);

  if (!configurations?.length) {
    return null;
  }

  return configurations.map(config => (
    <ProductConfigurationCard
      key={config?.detIdentifier}
      configuration={config}
      detModelKey={detModelKey}
      productPrice={{
        modelConfiguration,
        isFetching,
        hasError,
      }}
    />
  ));
};

MarineProductConfigurations.propTypes = componentProps;

export default MarineProductConfigurations;
