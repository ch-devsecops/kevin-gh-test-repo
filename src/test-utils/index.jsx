import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import { BrowserRouter } from 'react-router-dom';
import { OptionalVideoModalContextProvider } from '@honda-canada/design-system-react';
import UserLocationProvider from '@honda-canada/user-location';
import { configureAxe } from 'jest-axe';
import ModelExplorationContextProvider from '../components/ModelExplorationContext';
import ModelFiltersContextProvider from '../components/ModelFiltersContext';
import { MockFields, MockRendering } from './mockFields';
import MockTheme from './mockTheme';
import 'intersection-observer';
import { ACURA_THEME_NAME } from '../utils/constants';

const generateJssWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ACURA_THEME_NAME },
          ...context?.sitecoreContext,
        }}
        componentFactory={{}}
      >
        <BrowserRouter>
          <UserLocationProvider>
            <ThemeProvider theme={context?.theme || MockTheme}>
              <OptionalVideoModalContextProvider>
                <ModelExplorationContextProvider>
                  <ModelFiltersContextProvider>{children}</ModelFiltersContextProvider>
                </ModelExplorationContextProvider>
              </OptionalVideoModalContextProvider>
            </ThemeProvider>
          </UserLocationProvider>
        </BrowserRouter>
      </SitecoreContext>
    );
  };
/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in an typical JSS app's context providers.
 * Note that you can include a `context` object to set or override values
 * for SitecoreContext and ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.context] - default values for SitecoreContext, ThemeProvider
 * @param {string} [options.context.sitecoreContext] - value for SitecoreContext
 * @param {string} [options.context.theme] - value for ThemeProvider
 * @example
 * const { getByText } = jssRender(
 *  <MyComponent />, { context: { sitecoreContext: { language: 'fr' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const jssRender = (component, options = {}) => {
  const { context } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.context;

  return render(component, {
    wrapper: generateJssWrapper(context),
    ...options,
  });
};

const generateWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <BrowserRouter>
        <ThemeProvider theme={context?.theme || MockTheme}>
          <OptionalVideoModalContextProvider>{children}</OptionalVideoModalContextProvider>
        </ThemeProvider>
      </BrowserRouter>
    );
  };

/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in typical design-system context providers.
 * Note that you can include a `theme` object to set or override values
 * for ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.theme] - default value for ThemeProvider
 * @example
 * const { getByText } = render(
 *  <MyComponent />, { theme: { space: { wrapperGap: '15px' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const customRender = (component, options = {}) => {
  const { theme } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.theme;

  return render(component, {
    wrapper: generateWrapper(theme),
    ...options,
  });
};

// disable accessibility checks for errors that are usually trigged by content mistakes
const axe = configureAxe({
  rules: {
    'duplicate-id': { enabled: false },
    'heading-order': { enabled: false },
  },
});

export * from 'jest-axe';
export { axe };
export * from '@testing-library/react';
export { customRender as render, jssRender };
export { MockFields, MockRendering, MockTheme };
