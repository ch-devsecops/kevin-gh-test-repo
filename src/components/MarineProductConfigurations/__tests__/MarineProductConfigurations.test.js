import React from 'react';
import { jssRender, cleanup, axe } from 'test-utils';
import MarineProductConfigurations from '..';

afterEach(cleanup);

const mockData = require('../../../../sample-mock-data/routes/marine-product-configurations/en.json');
const emptyMockData = require('../../../../sample-mock-data/routes/marine-product-details-card/empty.json');

describe('<MarineProductConfigurations>', () => {
  test('should render correctly', () => {
    const { container } = jssRender(<MarineProductConfigurations {...mockData} />);

    expect(container).toMatchSnapshot();
    expect(container.children[0]).toBeTruthy();
  });

  test('should render correctly with empty mock data', () => {
    const { container } = jssRender(<MarineProductConfigurations {...emptyMockData} />);

    expect(container).toMatchSnapshot();
    expect(container.children[0]).toBeUndefined();
  });

  test('should return nulll if there is no fields', () => {
    const { container } = jssRender(<MarineProductConfigurations fields={null} />);
    expect(container).toMatchSnapshot();
  });

  test('should have no accessibility violations', async () => {
    const { container } = jssRender(<MarineProductConfigurations {...mockData} />);

    expect(await axe(container)).toHaveNoViolations();
  });
});
