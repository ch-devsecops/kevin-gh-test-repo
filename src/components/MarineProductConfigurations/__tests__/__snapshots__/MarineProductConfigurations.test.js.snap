// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MarineProductConfigurations> should render correctly 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 sc-1k44dfv-0 iyLtwN"
    data-testid="configuration-card"
  >
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 hGpHnT"
    >
      <h5
        class="sc-1k44dfv-0 sc-1k44dfv-0 hvDGYh"
        data-testid="configuration-name"
      >
        2_3DHSCHC
      </h5>
      <p
        class="sc-1k44dfv-0 dhqvJA"
        color="typographyDefault"
        data-testid="configuration-tagline"
        font-family="default"
        font-weight="regular"
      >
        Single cylinder, 15" short shaft, manual start, tiller throttle with clutch, integrated fuel tank
      </p>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 gAKfoF"
    >
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 icmvqr"
      />
    </div>
  </div>
  <div
    class="Box-sc-1bsf784-0 sc-1k44dfv-0 iyLtwN"
    data-testid="configuration-card"
  >
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 hGpHnT"
    >
      <h5
        class="sc-1k44dfv-0 sc-1k44dfv-0 hvDGYh"
        data-testid="configuration-name"
      >
        2_3DHSCHC
      </h5>
      <p
        class="sc-1k44dfv-0 dhqvJA"
        color="typographyDefault"
        data-testid="configuration-tagline"
        font-family="default"
        font-weight="regular"
      >
        Single cylinder, 15" short shaft, manual start, tiller throttle with clutch, integrated fuel tank
      </p>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 gAKfoF"
    >
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 icmvqr"
      />
    </div>
  </div>
</div>
`;

exports[`<MarineProductConfigurations> should render correctly with empty mock data 1`] = `<div />`;

exports[`<MarineProductConfigurations> should return nulll if there is no fields 1`] = `<div />`;
